mempool/reactor.go

```diff
@@ -213,6 +216,7 @@ func (memR *Reactor) broadcastTxRoutine(peer p2p.Peer) {
 				time.Sleep(PeerCatchupSleepIntervalMS * time.Millisecond)
 				continue
 			}
+			entry = nil
```

Comment:
Shouldn't this line be after the closing bracket of `if !memR.isSender(...`? Otherwise, if the peer is a sender, it will keep looping with the same entry, instead of fetching the next one.